### PR TITLE
(client) Implement the STIX malware-analysis object (#1565)

### PR DESCRIPTION
**Pull request by yassine-ouaamou**
### Proposed changes

*
*

### Related issues
[issue/1565](https://github.com/OpenCTI-Platform/opencti/issues/1565)

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

[STIX Malware Analysis SDO spec](https://docs.oasis-open.org/cti/stix/v2.1/cs01/stix-v2.1-cs01.html#_6hdrixb3ua4j)

### CODE REVIEW

File: pycti/entities/opencti_malware_analysis.py [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+                x_opencti_stix_ids=stix_object["x_opencti_stix_ids"]
+                if "x_opencti_stix_ids" in stix_object
+                else None,
+                objectOrganization=stix_object["granted_refs"]
+                if "granted_refs" in stix_object
+                else None,
+                update=update,
+            )
+        else:
+            LOGGER.error("[opencti_attack_pattern] Missing parameters: stixObject")
```
Comment (SarahBocognano) [inline pycti/entities/opencti_malware_analysis.py:460]: "[opencti_attack_pattern]" need to be [opencti_malware_analysis]

File: pycti/utils/opencti_stix2.py [hunk 2]
```diff
@@ -1623,31 +1624,38 @@ def prepare_export(
...
+
+                else:
                     key = (
                         stix_nested_ref_relationship["relationship_type"]
                         .replace("obs_", "")
                         .replace("-", "_")
                         + "_refs"
                     )
-                    if key in entity:
+                    if key in entity and isinstance(entity[key], list):
```
Comment (SarahBocognano) [inline pycti/utils/opencti_stix2.py:1651]: the isinstance is not very pretty. camelCase maybe ?

Comment (yassine-ouaamou) [reply pycti/utils/opencti_stix2.py:1651]: isinstance is a built-in Python function

File: pycti/utils/opencti_stix2.py [hunk 3]
```diff
@@ -1623,31 +1624,38 @@ def prepare_export(
         )
         for stix_nested_ref_relationship in stix_nested_ref_relationships:
             if "standard_id" in stix_nested_ref_relationship["to"]:
-                if MultipleRefRelationship.has_value(
+                # dirty fix because the sample and operating-system ref are not multiple for a Malware Analysis
```
Comment (Kedae) [inline pycti/utils/opencti_stix2.py:1627]: Still needed ?

Comment (yassine-ouaamou) [reply pycti/utils/opencti_stix2.py:1627]: Yes! Agreed with @richard-julien to do it this way for the moment

**Review by Kedae — APPROVED** [review]

**Pull request merged**